### PR TITLE
feat: detect nativeUserMode will be enabled so don't wait for keycloak

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

### What does this PR do?
`nativeUserMode` will be enabled by default on openshift with devworkspaces by https://github.com/eclipse-che/che-operator/pull/978. We need to reflect this in chectl so it does not wait for keycloak deployment.

Because `auth.nativeUserMode` is dynamicaly changed by che-operator, we don't know the value at the moment of listing the tasks. Current implementation gets and checks the current value in Keycloak deploy task. If we're deploying in `nativeUserMode: true`, the task is skipped (see the screenshot).

### Screenshot/screencast of this PR

Looks like this now:
20210805_070840_590x299_scrot

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20203

### How to test this PR?

 1. deploy with `./bin/run server:deploy --platform=openshift --installer=operator --workspace-engine=dev-workspace --che-operator-image=quay.io/mvala/che-operator:gh20203-nativeUserModeEnable` 
 2. chectl should complete process successfully

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
`Minikube E2E / e2e update tests (pull_request) ` is expected to fail (https://github.com/eclipse/che/issues/20235)

File: src/tasks/installers/common-tasks.ts [hunk 1]
```diff
@@ -155,6 +155,11 @@ export function createEclipseCheCluster(flags: any, kube: KubeHelper): Listr.Lis
   }
 }
 
+function shouldNotDeployKeycloak(flags: any, cr: any): boolean {
+  return cr.spec.auth.externalIdentityProvider || cr.spec.auth.nativeUserMode ||
+    (isOpenshiftPlatformFamily(flags.platform) && cr.spec.devWorkspace.enable)
+}
+
```
Comment (tolusha) [inline src/tasks/installers/common-tasks.ts:162]: Yes, that's a common issue when operator modifies checluster and chectl is not aware of that.
But in this case it leads to another problem. If chectl is used to deploy a previous version of Eclipse Che (--version flag) then keycloak deployment won't be handled. 

So, I suggest to modify [1] to pass some function to check if `nativeUserMode` is on and skip a task in this case.

[1] https://github.com/che-incubator/chectl/blob/main/src/tasks/kube.ts#L31
[2] https://github.com/che-incubator/chectl/blob/main/src/tasks/che.ts#L257
[3] https://github.com/che-incubator/chectl/blob/main/src/tasks/che.ts#L96

Comment (sparkoo) [reply src/tasks/installers/common-tasks.ts]: Done

File: src/tasks/che.ts [hunk 2]
```diff
@@ -76,6 +76,14 @@ export class CheTasks {
     this.cheDeploymentName = flags['deployment-name']
   }
 
+  async skipKeycloakDeploy(): Promise<boolean> {
+    const cheCluster = await this.kube.getCheCluster(this.cheNamespace)
+    if (!cheCluster) {
+      return false
+    }
+    return !(cheCluster.spec.auth.nativeUserMode == null || cheCluster.spec.auth.nativeUserMode == false);
```
Comment (tolusha) [inline src/tasks/che.ts:84]:
```suggestion
    return cheCluster.spec.auth.nativeUserMode
```

**Comment by tolusha** [comment]
Pls fix all errors 
https://github.com/che-incubator/chectl/pull/1502/files

**Comment by sparkoo** [comment]
> Pls fix all errors
> https://github.com/che-incubator/chectl/pull/1502/files

done

**Comment by sparkoo** [comment]
rebased so PR checks are happy

**Comment by sparkoo** [comment]
@tolusha CI seems happy. can you please rereview ?

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
`Minikube E2E / e2e update tests` is expected to fail

Pls merge.

**Pull request merged**